### PR TITLE
Cloud 1125 aks static ip

### PR DESCRIPTION
**Pull request by lee-baines**
Jira issue? CLOUD-1125
Release 6.5.0 backport required? Yes
6.5.0 doc changes needed? Yes
7.0.0 doc changes needed? Yes (David/Shankar lets catch up on this)
Required README updates made? No

- Enabled the scripts to use a static IP created outside of the Cluster resource group.  This is to prevent deletion of IP when Cluster is brought down.  Permission needs to be provided to the Service Principal for this to work.

- I've altered the scripts so providing a Service Principal is optional as this is created by default by AKS.

- Created AKS ingress script as was previously referencing EKS ingress script before.

### CODE REVIEW

**Comment by dgoldssfo** [comment]
Release 6.5.0 backport required? No(I don't think so)
6.5.0 doc changes needed? No

Should be yes for both, Lee.

File: bin/aks-create-cluster.sh [hunk 1]
```diff
@@ -25,6 +25,8 @@ echo -e "\tNode VM Size = ${AKS_NODE_VM_SIZE}"
 echo -e "\tNode Disk Size = ${AKS_NODE_OSDISK_SIZE}"
 echo -e "\tDefault Namespace = ${AKS_CLUSTER_NS}"
 echo -e "\tMonitoring Namespace = ${AKS_MONITORING_NS}"
+echo -e "\tStatic IP = ${AKS_INGRESS_IP}"
+echo -e "\tStatic IP Resource Group = ${AKS_IP_RESOURCE_GROUP_NAME}"
```
Comment (dgoldssfo) [inline bin/aks-create-cluster.sh]: echo -e "\tIngress Controller IP = ${GKE_INGRESS_IP}"

Wajih used this string in gke-create-cluster.sh. Either is OK, but could you make them consistent (either change this or change gke-create-cluster.sh) for a more polished look and feel?

File: bin/aks-create-cluster.sh [hunk 2]
```diff
@@ -35,20 +37,26 @@ case "${choice}" in
...
 CREATOR="${USER:-unknown}"
 # Labels can not contain dots that may be present in the user.name
 CREATOR=$(echo $CREATOR | sed 's/\./_/' | tr "[:upper:]" "[:lower:]")
 
-# Check first to see if SPN exists
-az ad sp show --id ${AKS_SERVICE_PRINCIPAL} > /dev/null 2>&1
-if [ $? -ne 0 ]; then
-    echo "Service Principal \"${AKS_SERVICE_PRINCIPAL}\" not found.  Please create first"
-    exit 1
+# Check first to see if Service Principal has been provide and whether it exist
```
Comment (dgoldssfo) [inline bin/aks-create-cluster.sh]: s/to see if Service Principal has been provide/to see if a Service Principal has been provided/

File: bin/aks-create-cluster.sh (hunk 1 above)
Comment (lee-baines) [reply bin/aks-create-cluster.sh]: ah yes, missed that.

File: bin/aks-create-cluster.sh (hunk 2 above)
Comment (lee-baines) [reply bin/aks-create-cluster.sh]: done

File: bin/aks-create-cluster.sh (hunk 1 above)
Comment (lee-baines) [reply bin/aks-create-cluster.sh]: done

**Pull request merged**